core/lib/storage/src/db.rs

```diff
@@ -304,7 +308,13 @@ impl<CF: NamedColumnFamily> RocksDB<CF> {
 
     pub fn with_options(path: &Path, options: RocksDBOptions) -> Result<Self, rocksdb::Error> {
         let caches = RocksDBCaches::new(options.block_cache_capacity);
-        let db_options = Self::rocksdb_options(None, None);
+        let mut db_options = Self::rocksdb_options(None, None);
+        let max_open_files = if let Some(non_zero) = options.max_open_files {
+            i32::try_from(non_zero.get()).ok().unwrap_or(i32::MAX)
```

Comment:
Nit: You can just use `i32::try_from(non_zero.get()).unwrap_or(i32::MAX)`; converting `Result` to an `Option` isn't necessary.